File: PPJ/własne programy/zad z grzeg 16 ppj/zad4/src/com/company/Osoba.java
package com.company;

public class Osoba {
    public String imie;
    public String nazwisko;
    public int rokUrodzenia;

    Osoba(String imie, String nazwisko, int rokUrodzenia){
        this.imie=imie;
        this.nazwisko=nazwisko;
        this.rokUrodzenia=rokUrodzenia;
    }
    public void show(){
        System.out.println(this.imie);
        System.out.println(this.nazwisko);
        System.out.println(this.rokUrodzenia);
    }
}

File: MAS/MAS_s20960_MP3/Wieloaspektowe/Osoba.cs
using MAS_s20960_MP3.Overlapping;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MAS_s20960_MP3.Wieloaspektowe
{
    public enum Sex
    {
        male,
        female
    }

    internal abstract class Osoba
    {
        public string Imie { get; set; }
        public string Nazwisko { get; set; }
        public string Panienskie { get; set; }
        public bool Wojsko { get; set; }
        public Sex sex { get; set; }

        public Osoba(string imie, string nazwisko, bool wojsko)
        {
            sex=Sex.male;
            Imie = imie;
            Nazwisko = nazwisko;
            Wojsko = wojsko;
        }

        public Osoba(string imie, string nazwisko, string panienskie)
        {
            sex = Sex.female;
            Imie = imie;
            Nazwisko = nazwisko;
            Panienskie = panienskie;
        }



        public string urodzenie()
        {
            if (sex == Sex.female)
            {
                return "urodzone";
            }
            else
                throw new Exception("Aborcja");
        }

        public void pobór()
        {
            if (sex == Sex.male)
            {
                this.Wojsko=true;
            }
            else
                throw new Exception("To nie jest mężczyzna");
        }



    }
}
